src/main/java/genesis/Genesis.java

```diff
@@ -0,0 +1,184 @@
+        }
+
+        String description = parts[0];
+        String by = parts[1];
+
+        Deadline deadline = new Deadline(description, by);
+        tasks.add(deadline);
+
+        ConsolePrinter.onTaskAdded(deadline.getListDescription(), tasks.size());
+    }
+
+    private void handleEvent(String content) throws GenesisException {
+        String[] parts = content.split(" /from | /to ", 3);
+        if (parts.length < 3) {
+            throw new GenesisException("Description for event is invalid");
+        }
+
+        String description = parts[0];
+        String from = parts[1];
+        String to = parts[2];
+
+        Event event = new Event(description, from, to);
+        tasks.add(event);
+
+        ConsolePrinter.onTaskAdded(event.getListDescription(), tasks.size());
+    }
+
+    private void handleDelete(String content) {
+        int index = Integer.parseInt(content) - 1;
+        Task task = tasks.remove(index);
+
+        ConsolePrinter.onTaskDelete(task.getListDescription(), tasks.size());
+    }
+
+
+    private void validateIndex(String[] contentArr) throws GenesisException {
+
+        if (contentArr.length < 2) {
+            throw new GenesisException("Index cannot be empty");
+        }
+    }
+
+    private void validateDescription(String[] contentArr) throws GenesisException {
+        if (contentArr.length < 2 || contentArr[1].isBlank()) {
+            throw new GenesisException("The description cannot be empty.");
+        }
+    }
+
+    public void askGenesis(String userInput) {
+        try {
+            ConsolePrinter.breakLine();
+
+            String[] contentArr = userInput.split(" ", 2);
+            String command = contentArr[0];
+
+            switch (command) {
+            case "list": {
+                handleListTasks();
+                return;
+            }
+            case "mark": {
+                validateIndex(contentArr);
+                handleMarkTask(contentArr[1]);
+                break;
+            }
+            case "unmark": {
+                validateIndex(contentArr);
+                handleUnmarkTask(contentArr[1]);
+                break;
+            }
+            case "todo": {
+                validateDescription(contentArr);
+                handleTodo(contentArr[1]);
+                break;
+            }
+            case "deadline": {
+                validateDescription(contentArr);
+                handleDeadline(contentArr[1]);
+                break;
+            }
+            case "event": {
+                validateDescription(contentArr);
+                handleEvent(contentArr[1]);
+                break;
+            }
+            case "delete": {
+                validateIndex(contentArr);
+                handleDelete(contentArr[1]);
+                break;
+            }
+            default: {
+                throw new UnknownCommandException("I'm sorry, but I don't know what that means :-(");
+            }
+            }
```

Comment:
spacing/indentation issues?

Reply:
I think its because of me using case statements enclosed in braces. I will get this fixed.